libs/features/personalization/personalization.js

```diff
@@ -105,6 +105,17 @@ export const preloadManifests = ({ targetManifests = [], persManifests = [] }) =
 
 export const getFileName = (path) => path?.split('/').pop();
 
+export const isDelayedModal = (url) => {
+  const delayedModalRegExp = /\/.*?#.*?:delay=[1-9][0-9]*/; // matches 'anytext/#anytext:delay=anynumbers(followed by any text)
+  let isDM = false;
+  try {
+    isDM = delayedModalRegExp.test(url);
```

Comment:
When does regex `test` throw an error?  I'm not aware of any value that would cause it to.

Reply:
The line break can cause the error, e.g.
```
const delayedModalRegExp = /\/.*?#.*?:delay=[1-9][0-9]*/;
delayedModalRegExp.test('This is a string
with a line break.')
// Uncaught SyntaxError: Invalid or unexpected token
```
I know it's pretty unlikely this may happen, but authors may copy-paste such string into Excel from somewhere else, so I decided to add this checking as we don't have any validation on the authoring side.

Reply:
Do you think there is a use case where a delayed modal would be triggered by a cta on the page?  I can't think of why that would ever be needed.  If that's the case, then I think we can move this logic directly into modal.js.

`function delayedModal` could directly add `el.classList.add('hide-block');` to the modal link.

Then we don't even need any support for `#hide`, and there's no code changes needed in personalization.js